Command: Add option to add/remove environment includes (#44285)
spack/spack: lib/spack/docs/environments.rst

     - /absolute/path/to/environment1
     - /absolute/path/to/environment2

> It is also possible to include concrete environments from the command line either by
> environment name or by path.

Reviewer: Not sure about final formatting but:

```suggestion
Or you can use the command line to include concrete environments using their name or path.
```

.. code-block:: console

   $ spack -e included_env include --concrete add environment1

>    $ spack -e included_env include --concrete add /absolute/path/to/environment1

Reviewer: TBH. I'm not a fan of having the `--concrete` option appear before the subcommand.  There is a precedent for having such options added to each subcommand to which they apply (e.g., see https://spack.readthedocs.io/en/latest/command_index.html#spack-mirror).

Once the ``spack.yaml`` has been updated you must concretize the environment to
get the concrete specs from the included environments.

> To list which concrete environments are included in the current active environment
> from the command line use the ``spack include --concrete list`` command.

Reviewer: Since the command also works for includes, perhaps something like the following (preferably under a separate subsection):

```suggestion
You can list includes for the active environment
at the command line using the
``spack include list`` command. For example,
```

Reviewer: I see the following after adding a `definitions.yaml` file to one of my environments:

```
$ spack include list
==> Included Configuration Scopes
==> 
	definitions.yaml ($spack/var/spack/environments/defs/definitions.yaml)
```

.. code-block:: console

   $ spack include --concrete list
   ==> Included Concrete Environments
           environment1 (/absolute/path/to/environment1)
           environment2 (/absolute/path/to/environment2)

> 

Reviewer: I'd like to see this as a separate subsection, such as `Listing environment includes`.

   ==> 0 installed packages

> It is also required to reconcretize to reflect the changes of a new environment being
> added or removed. For example if ``myenv`` were to be removed and a new environment

Reviewer: Perhaps something along the lines:

```suggestion
Changes to environment includes require reconcretizing the environment.
For example if ``myenv`` were to be removed and a new environment
```

``externalpython`` were added.

.. code-block:: console

>   $ spack env create myotherenv

Reviewer: Should this be `externalpython`?

  $ spack -e externalpython add python
  $ spack -e externalpython external find --not-buildable python
  ==> The following specs have been detected on this system and added to $spack/var/spack/environments/externalpython/spack.yaml
  python@3.8.10
  $ spack -e externalpython concretize

  $ spack -e included_env include --concrete remove myenv
  $ spack -e included_env include --concrete add externalpython

>   $ spack -e included_env concretize

Reviewer: This doesn't quite fit the subsection heading goal of explaining how to ensure the environment gets the latest versions of included environments.  Perhaps it should be a separate subsection about updating includes (e.g., `Updating environment includes`)?

schemes). Spack-specific, environment and user path variables may be
used in these paths. See :ref:`config-file-variables` for more information.

> It is possible to add or remove configurations to the current active environment

Reviewer: Not sure about final formatting, but:

```suggestion
You can add and remove configuration files in the current active environment
```

from the command line. When includes are added they will be added with the highest
precendence. In the case of multiple includes being added the precendence increases
left to right in the same way ``--config-scope`` arguments are applied.

> 

Reviewer: ```suggestion
Be sure to use the path of the includes on the command line.

```

Reviewer: Above could be a hint.

The reason I bring it up is that `spack include list` will give a filename for a file (along with the absolute path). So I can see someone trying to use the filename and wondering why that doesn't work.
